### DISCUSSION TITLE
No audio recording on new Pi install

### DISCUSSION THREAD

**Discussion by GeekNJ** [mcguirepr89/BirdNET-Pi#437] [Q&A] [answered]
Hey all! I'm familiar with PIs as I have a bunch scattered around the house doing various things and saw the article on birdnetpi . We have feeders and lots of birds so I fired up a headless pi 4 w/ 2GB for this and installed the latest OS Lite on a SD card. I have a USB mic attached but I'm seeing any audio when launching the birdnet web interface.

I went through setup of the USB microphone and running this command records audio from the USB mic and plays it back through the headphone jack:
arecord -d 3 test.wav && aplay test.wav

I used the Lite OS here so no desktop but command line works fine. Not exactly sure if it's a mono or stereo mic so I tried both 1 and 2 for channels in advance settings but that didn't matter.

Any thoughts on how to troubleshoot? Some more details below. Many thanks!

David

birdnetpi:~/BirdNET-Pi$ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 189a:2019  USB Microphone
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

birdnetpi:~/BirdNET-Pi$ arecord -l
**** List of CAPTURE Hardware Devices ****
card 1: Microphone [USB Microphone], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
  
Manually edited /usr/share/alsa/alsa.conf and set defaults.pcm.card 1
Also have .asoundrc set.
birdnetpi:~/BirdNET-Pi$ cat ~/.asoundrc
pcm.!default {
  type asym
  capture.pcm "mic"
}
pcm.mic {
  type plug
  slave {
    pcm "hw:1,0"
  }
}

System all updated:

birdnetpi:~/BirdNET-Pi$ sudo apt update
Hit:1 http://deb.debian.org/debian bullseye InRelease
Hit:2 http://deb.debian.org/debian bullseye-updates InRelease                                                    
Hit:3 http://security.debian.org/debian-security bullseye-security InRelease                                     
Hit:4 http://archive.raspberrypi.org/debian bullseye InRelease                                                   
Get:5 https://dl.cloudsmith.io/public/caddy/stable/deb/debian any-version InRelease [7,491 B]
Fetched 7,491 B in 1s (6,378 B/s)    
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.

birdnetpi:~/BirdNET-Pi$ sudo apt full-upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

**Comment by mcguirepr89** [comment, marked as answer]
Before manually changing the `alsa.conf` file, did you test to see if the system would have worked with the default settings (`pulseaudio` abstracting all of those settings as `default`)?

**Reply by GeekNJ** [reply]
Yes. I only looked up changes after I noticed it wasn't recording showing anything with the out of the box install. When I did the initial birdnet install, the microphone was not connected at that time. 

I can do a fresh install again from an OS image with the mic installed on first boot and birdnetpi install if that is desired.

**Reply by mcguirepr89** [reply]
Yes, please do a clean install and then, based on the errors we can see from log entries, we can adjust some settings to get the microphone working -- based on the output from your alsa commands, PulseAudio ought to be able to wrap the mic without any issue, but we'll certainly see.

After a clean install, before making any adjustments, if the recording/analysis is not working, I'll kindly ask that you send whatever you can see in the "View Logs" page, after which I will likely request more specific log information.

Thank you and :crossed_fingers:

**Reply by GeekNJ** [reply]
Clean install done with mic installed from the start and all good with the basic steps and it showing info immediately on the spectrogram:

Re-imaged 64-bit OS Lite onto SD card
sudo apt update
sudo apt full-upgrade
sudo apt clean
sudo reboot
curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Pi/main/newinstaller.sh | bash

Thanks so much and I look forward to seeing what it picks up at our feeders.
